guides/v2.2/javascript-dev-guide/widgets/widget-row-builder.md

```diff
@@ -0,0 +1,267 @@
+    "remEventSelector":"a",
+    "btnRemoveSelector":".action.remove",
+    "maxRows":"5"
+});``
+```
+
+Where:
+
+- `#row-builder` is the selector of the element which will be the RowBuilder.
+
+Phtml template file examples using script:
+
+```html
+<script>
+    require([
+        'jquery',
+        'rowBuilder'
+    ], function ($) {
+        'use strict';
+        
+        $("#row-builder").rowBuilder({
+              "rowTemplate":"#row-template",
+              "rowContainer":"#row-container",
+              "rowParentElem":"<div></div>",
+              "remEventSelector":"a",
+              "btnRemoveSelector":".action.remove",
+              "maxRows":"5"
+        });
+    });
+</script>
+```
+
+## Options
+
+The PopupWindow widget has the following options:
+
+-  [rowTemplate](#rowtemplate)
+-  [rowContainer](#rowcontainer)
+-  [rowIndex](#rowindex)
+-  [rowCount](#rowcount)
+-  [rowParentElem](#rowparentelem)
+-  [rowContainerClass](#rowcontainerclass)
+-  [addRowBtn](#addrowbtn)
+-  [btnRemoveIdPrefix](#btnremoveidprefix)
+-  [btnRemoveSelector](#btnremoveselector)
+-  [rowIdPrefix](#rowidprefix)
+-  [additionalRowClass](#additionalrowclass)
+-  [addEventSelector](#addeventselector)
+-  [remEventSelector](#remeventselector)
+-  [hideFirstRowAddSeparator](#hidefirstrowaddseparator)
+-  [maxRows](#maxrows)
+-  [maxRowsMsg](#maxrowsmsg)
+
+### `rowTemplate`
+
+The row template selector.
+
+**Type**: String
+
+**Default value**: `'#template-registrant'`
+
+### `rowContainer`
+
+The row container selector.
+
+**Type**: String
+
+**Default value**: `'#registrant-container'`
+
+### `rowIndex`
+
+Row index used by the template rows.
+
+**Type**: Integer
+
+**Default value**: `0`
+
+### `rowCount`
+
+Row count.
+
+**Type**: Integer
+
+**Default value**: `0`
+
+### `rowParentElem`
+
+Row parent element.
+
+**Type**: String
+
+**Default value**: `'<li></li>'`
+
+### `rowContainerClass`
+
+Row container class.
+
+**Type**: String
+
+**Default value**: `'fields'`
+
+### `addRowBtn`
+
+The add row button selector.
+
+**Type**: String
+
+**Default value**: `'#add-registrant-button'`
+
+### `btnRemoveIdPrefix`
+
+Id prefix of a remove button.
+
+**Type**: String
+
+**Default value**: `'btn-remove'`
+
+### `btnRemoveSelector`
+
+The remove button selector.
+
+**Type**: String
+
+**Default value**: `'.btn-remove'`
+
+### `rowIdPrefix`
+
+Id prefix of a row element.
+
+**Type**: String
+
+**Default value**: `'row'`
+
+### `additionalRowClass`
+
+The additional row class is added to rows added after the first one.
+
+**Type**: String
+
+**Default value**: `'add-row'`
+
+### `addEventSelector`
+
+Default selectors for add element of a template.
+
+**Type**: String
+
+**Default value**: `'button'`
+
+### `remEventSelector`
+
+Default selectors for remove markup elements of a template.
+
+**Type**: String
+
+**Default value**: `'a'`
+
+### `hideFirstRowAddSeparator`
+
+This option allows adding first row delete option and a row separator.
```

Comment:
Please, adjust this explanation. Technically, if we split the purpose of this option into parts:
- allows adding first row delete option
- allows adding a row separator

This explanation confuses a little bit. So we need to simplify it slightly. 

Thank you.

Reply:
@rogyar thank you for noticing it.
I've adjusted the explanation of the **hideFirstRowAddSeparator** option.
This option allows removing the additional row class and remove link for the first row.
Source code: https://github.com/magento/magento2/blob/2.2/app/code/Magento/Theme/view/frontend/web/js/row-builder.js#L151
https://github.com/magento/magento2/blob/2.3/app/code/Magento/Theme/view/frontend/web/js/row-builder.js#L151